CRM/Contribute/Form/UpdateBilling.php

```diff
@@ -75,12 +73,8 @@ public function preProcess() {
     if ((!$this->_crid && !$this->_coid && !$this->_mid) || (!$this->_subscriptionDetails)) {
       CRM_Core_Error::fatal('Required information missing.');
     }
-    if (!CRM_Core_Permission::check('edit contributions')) {
-      if ($this->_subscriptionDetails->contact_id != $this->getContactID()) {
-        CRM_Core_Error::statusBounce(ts('You do not have permission to cancel subscription.'));
-      }
-      $this->_selfService = TRUE;
-    }
+
+    $this->isSelfService();
```

Comment:
shouldn't we move this from here to the buildQuickForm & instead of if($this->selfService) it would be if ($this->isSelfService())

Reply:
ok done i think your right in this case

Reply:
I'm also right about whether 'your' should have an appostrophe :-)